src/Func.ls

```diff
@@ -12,8 +12,15 @@ fix = (f) ->
 
 over = (f, g, x, y) --> f (g x), (g y)
 
-#? memoize, wrap
+memoize = (f) ->
+  memo = {}
+  ->
+    args = Array::slice.call &, 0
+    key = args.join ''
```

Comment:
Even a pair of parallel arrays with `in` might work here. I will note that,
in an object, `obj[1]` and `obj['1']` are equivalent. This would be the
other problem.
On Jan 2, 2015 6:07 PM, "John Wright" <notifications@github.com> wrote:

> In src/Func.ls
> <https://github.com/gkz/prelude-ls/pull/72#discussion-diff-22426241>:
>
> > @@ -12,8 +12,15 @@ fix = (f) ->
> >
> >  over = (f, g, x, y) --> f (g x), (g y)
> >
> > -#? memoize, wrap
> > +memoize = (f) ->
> > +  memo = {}
> > +  ->
> > +    args = Array::slice.call &, 0
> > +    key = args.join ''
>
> I pinched this technique from lodash, but you're right. Unfortunately,
> though, Map isn't available in the stable version of node. I'll see if I
> can come up with something a little better.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gkz/prelude-ls/pull/72/files#r22426241>.
>